package.json

```diff
@@ -2,30 +2,32 @@
   "name": "docker-progress",
   "version": "4.0.3",
   "description": "This module provides additional progress mechanics on top of dockerode",
-  "main": "index.js",
   "repository": {
     "type": "git",
     "url": "git@github.com:balena-io-modules/docker-progress.git"
   },
   "author": "Pagan Gazzard <page@balena.io>",
-  "types": "index.d.ts",
+  "main": "build/index.js",
+  "types": "build/index.d.ts",
+  "files": [
+    "build/"
+  ],
   "scripts": {
-    "lint": "balena-lint index.coffee legacy.coffee",
+    "clean": "rimraf build",
+    "build": "npm run clean && npm run lint && tsc",
+    "lint": "balena-lint -e ts --typescript --fix lib",
     "test": "npm run lint",
-    "prepublish": "npm run test && coffee -c index.coffee && coffee -c legacy.coffee"
+    "prepublishOnly": "npm run build"
```

Comment:
I think this should be a `prepare` btw @pdcastro

Reply:
Should it? I gather that `prepare` is a closer match to `prepublish` (which is deprecated), but `prepare` is  also automatically executed after `npm install`. Do we want this? Most repos don't trigger build / compilation (`tsc`) by simply running `npm install`, right?